src/core/base64.cpp

```diff
@@ -54,4 +54,134 @@ QVariant decode( const QString & _b64, QVariant::Type _force_type )
 }
 
 
+//TODO C++20: It *may* be possible to make the following functions constepxr given C++20's constexpr std::string.
+
+/**
+ * @brief Base64 encodes data.
+ * @param data
+ * @return std::string containing the Base64 encoded data.
+ */
+std::string encode(std::string_view data)
+{
+	if (data.empty()) { return ""; }
+
+	// base64 encoded string
+	std::string result;
+
+	// number of chunks to process + padding
+	auto [numChunks, numTrailingBytes] = std::div(data.length(), numBytesPerChunk);
```

Comment:
The `std::div` function returns a `div_t` object whose `quot` and `rem` members are not guaranteed to be in any particular order ([see link](https://en.cppreference.com/w/cpp/numeric/math/div)), so using a structured binding here would result in unpredictable behavior.

Reply:
Fixed! Thanks for showing that. I wasn't aware `div_t` wasn't in a guaranteed order! 😩